rero_mef/marctojson/do_idref_places.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+#
+# RERO MEF
+# Copyright (C) 2020 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+"""Marctojsons transformer for IdRef records."""
+
+import contextlib
+from datetime import datetime, timezone
+
+from rero_mef.marctojson.helper import build_string_from_field, \
+    build_string_list_from_fields
+
+
+class Transformation(object):
+    """Transformation MARC21 to JSON for Idref concept."""
+
+    def __init__(self, marc, logger=None, verbose=False, transform=True):
+        """Constructor."""
+        self.marc = marc
+        self.logger = logger
+        self.verbose = verbose
+        self.json_dict = {}
+        if transform:
+            self._transform()
+
+    def _transform(self):
+        """Call the transformation functions."""
+        if fields_008 := self.marc.get_fields('008'):
+            for func in dir(self):
+                if func.startswith('trans'):
+                    func = getattr(self, func)
+                    func()
+        else:
+            msg = 'No 008'
+            if self.logger and self.verbose:
+                self.logger.warning(f'NO TRANSFORMATION: {msg}')
+            self.json_dict = {'NO TRANSFORMATION': msg}
+            self.trans_idref_pid()
+
+    @property
+    def json(self):
+        """Json data."""
+        return self.json_dict or None
+
+    def trans_idref_pid(self):
+        """Transformation identifier from field 001."""
+        if self.logger and self.verbose:
+            self.logger.info('Call Function', 'trans_idref_pid')
+        if field_001 := self.marc.get_fields('001'):
+            self.json_dict['pid'] = field_001[0].data
+            self.json_dict['type'] = 'bf:Place'
+
+    def trans_idref_bnf_type(self):
+        """Transformation bnf_type from field 008."""
+        if self.logger and self.verbose:
+            self.logger.info('Call Function', 'trans_idref_bnf_type')
+        if field_008 := self.marc.get_fields('008'):
+            self.json_dict['bnf_type'] = 'sujet Rameau'
+            if field_008[0].data == 'Tf8':
+                self.json_dict['bnf_type'] = 'genre/forme Rameau'
```

Comment:
I think this function is not needed for places. Places cannot be genre/forme Rameau, I think they will always be "sujet Rameau".